torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -4839,15 +4839,19 @@ Error PyTorchModelLoader::loadPermute(const torch::jit::Node *ptNode) {
 Error PyTorchModelLoader::loadTo(const torch::jit::Node *ptNode) {
   auto inputs = ptNode->inputs();
   auto outputs = ptNode->outputs();
-  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 5, outputs, 1));
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, -5, outputs, 1));
 
   glow::NodeValue input;
   ASSIGN_VALUE_OR_RETURN_ERR(input,
                              getGlowNodeValueForValue(inputs[ToInputs::input]));
 
   int32_t dtype;
-  ASSIGN_VALUE_OR_RETURN_ERR(
-      dtype, iValToInt(getGlowIValueForValue(inputs[ToInputs::dtype])));
+  if (inputs.size() == 5 || inputs.size() == 8) {
```

Comment:
Actually when inputs.size() == 7 we also support this. However, do you have any known use case of aten::to which has 7 or 8 params? If it there is then a better solution is to re-write here for each case instead of this, or if it is not maybe we can just keep it this way.

Reply:
Hi @zrphercule, it seems that using `tensor.to(device)` generates the 8 parameter variant

```
aten::to(%1927, %1822, %1803, %1802, %1820, %1821, %1821, %1820)
```

I am not sure about the 7 parameter variant, but I can add it to the check and allow it too?

Regarding rewriting for each case, I am not sure if that's needed since the extra parameters are not used and I don't think they are supported in glow.

Reply:
Hi @zrphercule, do you have any more comments on this?